modules/page-memory/src/main/java/org/apache/ignite/internal/pagememory/Storable.java

```diff
@@ -59,5 +67,27 @@ public interface Storable {
     /**
      * Returns I/O for handling this storable.
      */
-    IoVersions<? extends AbstractDataPageIo<?>> ioVersions();
+    IoVersions<DataPageIo> ioVersions();
+
+    /** Returns a byte buffer that contains binary tuple data. */
```

Comment:
Excuse me, what binary tuple? It's an abstraction, it should not depend on the implementation. There's clearly something wrong here.
What was wrong with `writeRowData` and `writeFragmentData` that you could have moved here?

Reply:
removed, moved `writeRowData `and `writeFragmentData` to storable implementations